docs/proposals/sysctl.md

```diff
@@ -0,0 +1,477 @@
+      for each socket. While large values are necessary for certain environments and applications, they lead to waste of resources
+      in the 90% case.
+  * some sysctls have a different error behavior, e.g.:
+    * creating a shared memory segment will fail immediately when `kernel.shmmax` is too small.
+
+      With a large `kernel.shmmax` default, the creation of a segment always succeeds, but the OOM killer will
+      do its job when a shared memory segment exceeds the memory request of the container.
+
+  The high values that could be set by the kubelet on launch might depend on the node's capacity and capabilities. But for
+  portability of workloads it is helpful to have a common baseline of sysctls settings one can expect on every node. The
+  kernel defaults (which are active if the kubelet does not change defaults) are such a (natural) baseline.
+
+## Analysis of Sysctls on the initial Whitelist
+
+- `kernel.shmall`, `kernel.shmmax`, `kernel.shmmni`: configure System V shared memory
+  * [x] **namespaced** in ipc ns
+  * [x] **accounted for** as user memory in memcg, using sparse allocation (like tmpfs)
+    uses [Resizable virtual memory filesystem](https://github.com/torvalds/linux/blob/master/mm/shmem.c)
+  * [x] hence **safe to customize**
+  * [x] **no application influence** with high values
+  * **defaults to** [unlimited pages, unlimited size, 4096 segments on today's kernels](https://github.com/torvalds/linux/blob/0e06f5c0deeef0332a5da2ecb8f1fcf3e024d958/include/uapi/linux/shm.h#L20). This make **customization practically unneccessary**.
+- `kernel.shm_rmid_forced`: enforce removal of shared memory segments on process shutdown
+  * [x] **namespaced** in ipc ns
+- `kernel.msgmax`, `kernel.msgmnb`, `kernel.msgmni`: configure System V messages
+  * [x] **namespaced** in ipc ns
+  * [ ] [temporarily **allocated in kmem** in a linked message list](http://lxr.linux.no/linux+v4.7/ipc/msgutil.c#L58), but **not accounted for** in memcg
+  * [ ] **defaults to** [8kb, 16384 messages, 32000 queues](http://lxr.linux.no/linux+v4.7/include/uapi/linux/msg.h#L75), **which might be too small** for certain applications
+  * [ ] arbitrary values [up to INT_MAX](http://lxr.linux.no/linux+v4.7/ipc/ipc_sysctl.c#L135). Hence, **potential DoS attack vector** against the host
+- `fs.mqueue.*`: configure POSIX message queues
+  * [x] **namespaced** in ipc ns
+  * [ ] uses the same [`load_msg`](http://lxr.linux.no/linux+v4.7/ipc/msgutil.c#L58) as System V messages, i.e. **no accounting**
+  * does [strict checking again rlimits](http://lxr.free-electrons.com/source/ipc/mqueue.c#L278) though
+  * [ ] **defaults to** [256 queues, max queue length 10, message size 8kb](http://lxr.free-electrons.com/source/include/linux/ipc_namespace.h#L102)
+  * [ ] can be customized via sysctls up to 64k max queue length, message size 16MB. Hence, **potential DoS attack vector** against the host
```

Comment:
What I mean here is that with that sysctl and arbitrary values you can probably take the node down by launching a pod.